File: src/workflows/workflowAlignmentEngine.ts
/**
 * Converted from Python: core/workflows/workflow_alignment_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function alignWorkflowRuntime(plan: any, state: any, execution: any): any {
  return {"objective": py.get(plan, "objective", ""), "steps_aligned": py.eq(py.len(py.get(plan, "steps", [])), py.get(execution, "completed_count", 0)), "state_step": py.get(state, "current_step", 0), "aligned": true, "bounded": true};
}

File: lib/src/workflows/workflow_alignment_engine.dart
// Port of core/workflows/workflow_alignment_engine.py

Map<String, dynamic> alignWorkflowRuntime(
  Map<String, dynamic> plan,
  Map<String, dynamic> state,
  Map<String, dynamic> execution,
) {
  final List<dynamic> steps = plan['steps'] is List
      ? List<dynamic>.from(plan['steps'] as List<dynamic>)
      : <dynamic>[];
  final int completedCount = execution['completed_count'] is int
      ? execution['completed_count'] as int
      : 0;

  return <String, dynamic>{
    'objective': plan['objective'] ?? '',
    'steps_aligned': steps.length == completedCount,
    'state_step': state['current_step'] ?? 0,
    'aligned': true,
    'bounded': true,
  };
}
